unstructured/partition/pdf_image/analysis/__init__.py

```diff
@@ -0,0 +1,169 @@
+import json
+import uuid
+from io import BytesIO
+from pathlib import Path
+from typing import Optional
+
+from unstructured import env_config
+from unstructured.partition.pdf_image.analysis.bbox_visualisation import (
+    AnalysisDrawer,
+    FinalLayoutDrawer,
+    LayoutDrawer,
+    OCRLayoutDrawer,
+    ODModelLayoutDrawer,
+    PdfminerLayoutDrawer,
+)
+from unstructured.partition.pdf_image.analysis.layout_dump import (
+    ExtractedLayoutDumper,
+    FinalLayoutDumper,
+    JsonLayoutDumper,
+    LayoutDumper,
+    ObjectDetectionLayoutDumper,
+    OCRLayoutDumper,
+)
+
+
+def _get_drawer_for_dumper(dumper: LayoutDumper) -> Optional[LayoutDrawer]:
+    """For a given layout dumper, return the corresponding layout drawer instance initialized with
+    a dumped layout dict.
+
+    Args:
+        dumper: The layout dumper instance
+
+    Returns:
+        LayoutDrawer: The corresponding layout drawer instance
+    """
+    if isinstance(dumper, ObjectDetectionLayoutDumper):
+        return ODModelLayoutDrawer(layout_dump=dumper.dump())
+    elif isinstance(dumper, ExtractedLayoutDumper):
+        return PdfminerLayoutDrawer(layout_dump=dumper.dump())
+    elif isinstance(dumper, OCRLayoutDumper):
+        return OCRLayoutDrawer(layout_dump=dumper.dump())
+    elif isinstance(dumper, FinalLayoutDumper):
+        return FinalLayoutDrawer(layout_dump=dumper.dump())
+    else:
+        raise ValueError(f"Unknown dumper type: {dumper}")
+
+
+def _generate_filename(is_image: bool):
+    """Generate a filename for the analysis artifacts based on the file type.
+    Adds a random uuid suffix
+    """
+    suffix = uuid.uuid4().hex[:5]
+    if is_image:
+        return f"image_{suffix}.png"
+    return f"pdf_{suffix}.pdf"
+
+
+def save_analysis_artifiacts(
+    *layout_dumpers: LayoutDumper,
+    is_image: bool,
+    analyzed_image_output_dir_path: str,
+    filename: Optional[str] = None,
+    file: Optional[BytesIO] = None,
+):
+    """Save the analysis artifacts for a given file. Loads some settings from
+    the environment configuration.
+
+    Args:
+        layout_dumpers: The layout dumpers to save and use for bboxes rendering
+        filename: The filename of the sources analyzed file (pdf/image)
+        analyzed_image_output_dir_path: The directory to save the analysis artifacts
+    """
+    if not filename:
+        filename = _generate_filename(is_image)
+    skip_bboxes = env_config.ANALYSIS_BBOX_SKIP
+    skip_dump_od = env_config.ANALYSIS_DUMP_OD_SKIP
```

Comment:
now you can't use this function without `env_config`, so it's not pure

Reply:
hmm right - I'll parametrize this and pass from the main partition function.